spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,155 @@
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+As this attestation is generated by a version control service or code review
+service, the signatures are from the service rather than the specific users
+issuing the review.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "name": "<REPO>",
+            "digest": { "gitCommit": "<HASH>" },
+            "uri": "<LINK TO REPO>"
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewers": ["<ResourceDescriptor>", ...],
+        "targetTip": "<ResourceDescriptor>",
+        "mergeBase": "<ResourceDescriptor>",
+        "reviewLink": "<LINK TO REVIEW>",
+        "reviewTime": "<TIMESTAMP>",
+        "annotations": {...}
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+`reviewers` _array of ResourceDescriptor_, _required_
+
+Indicates uniquely the identity of one or more reviewers. MUST NOT be used over
+the attestation's signature to validate the source of the attestation. This
+field enables scenarios where the attestation is issued and signed by the VCS
+review system. Each entry must include immutable information that is always
+mapped to the reviewer. The digest may be skipped in the ResourceDescriptor,
+using the name and URI fields to record the username and immutable ID.
+
+`targetTip` _ResourceDescriptor_, _required_
+
+Records the target location for the proposed changeset. In Git repositories, the
+field must contain the base Git reference (i.e., the base branch) AND the
+current commit ID at the tip of that branch.
+
+`mergeBase` _ResourceDescriptor_, _required_
+
+Records the base of the proposed changeset. In Git repositories, three way
+merges are performed using the tip of the topic branch, the tip of the target
+(or base) branch, and the merge base. In a fast-forward merge, the mergeBase is
+the same as tip of the target branch. Using the subject of the attestation, the
+`targetTip`, and the `mergeBase`, the set of changes reviewed can be inferred.
+
+TODO: this is becoming very Git specific.
+
+`reviewLink` _URI_, _optional_
+
+Contains a link to the full review. Useful to point to information that cannot
+be captured in the attestation.
+
+`reviewTime` _Timestamp_, _optional_
+
+Indicates the time of _last review_ creation. If the attestation is for a single
+review, this is equivalent to when that review was created. With multiple
+reviewers, the attestation is re-created for each new review and the time
+reflects that of the latest review.
```

Comment:
What happens to the old attestations that are superseded by the more recent review attestations? I'm wondering if we should add the ability to link to prior attestations related to the same subject PR to maintain full visibility into the review history.

Reply:
Related: https://github.com/in-toto/attestation/pull/151#discussion_r1330386183

I think it depends on the intended policies. I'm not sure we have policies / use cases that require full review history, but I'm not opposed to including the information. This is currently written with the admittedly simpler rule `k of n approve changes` in mind.

Reply:
For a v0.1, I think the `k out of n` assumption is actually fine. I would just make it explicit, if it isn't stated anywhere yet.